sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Plan.java

```diff
@@ -21,6 +22,8 @@
  * list of one or more error messages to be shown to the user,
  */
 public interface Plan extends ParentElement<Phase> {
+    void setErrors(List<String> errors);
```

Comment:
How about something that `extends Plan`, wraps the original Plan object, and just overrides the `getErrors()` call (and probably also the `getStatus()` call)? Then it'd be a bit more explicit that we're just adding errors onto the existing plan.

```
public class ErrorWrapper {
  public ErrorWrapper(Plan wrapMe, List<String> errors) {
    this.wrapMe = wrapMe;
    this.errors = errors;
  }

  @Override
  public List<String> getErrors() {
    List<String> tmp = new ArrayList<>();
    tmp.addAll(wrapMe.getErrors())
    tmp.addAll(errors);
    return tmp;
  }

  @Override
  public Status getStatus() {
    return ERROR;
  }

  [.. probably some boilerplate here ..]
}
```